bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3643,45 +3643,41 @@ void ReplicaImp::executeReadOnlyRequest(concordUtils::SpanWrapper &parent_span,
 
   uint16_t clientId = request->clientProxyId();
 
-  int error = 0;
-  uint32_t actualReplyLength = 0;
-  uint32_t actualReplicaSpecificInfoLength = 0;
-
+  int status = 0;
+  bftEngine::IRequestsHandler::ExecutionRequestsQueue accumulatedRequests;
+  accumulatedRequests.push_back(bftEngine::IRequestsHandler::ExecutionRequest{clientId,
+                                                                              static_cast<uint64_t>(lastExecutedSeqNum),
+                                                                              request->flags(),
+                                                                              request->requestLength(),
+                                                                              request->requestBuf(),
+                                                                              reply.maxReplyLength(),
+                                                                              reply.replyBuf()});
   {
     TimeRecorder scoped_timer(*histograms_.executeReadOnlyRequest);
-    error = bftRequestsHandler_.execute(clientId,
-                                        lastExecutedSeqNum,
-                                        request->flags(),
-                                        request->requestLength(),
-                                        request->requestBuf(),
-                                        reply.maxReplyLength(),
-                                        reply.replyBuf(),
-                                        actualReplyLength,
-                                        actualReplicaSpecificInfoLength,
-                                        span);
+    bftRequestsHandler_.execute(accumulatedRequests, request->getCid(), span);
   }
-
+  const IRequestsHandler::ExecutionRequest &single_request = accumulatedRequests.back();
+  status = single_request.outExecutionStatus;
   LOG_DEBUG(GL,
             "Executed read only request. " << KVLOG(clientId,
                                                     lastExecutedSeqNum,
                                                     request->requestLength(),
                                                     reply.maxReplyLength(),
-                                                    actualReplyLength,
-                                                    actualReplicaSpecificInfoLength,
-                                                    error));
-
+                                                    single_request.outActualReplySize,
+                                                    single_request.outReplicaSpecificInfoSize,
+                                                    status));
   // TODO(GG): TBD - how do we want to support empty replies? (actualReplyLength==0)
-  if (!error) {
-    if (actualReplyLength > 0) {
-      reply.setReplyLength(actualReplyLength);
```

Comment:
In order to minimize the code changes, let’s just set values to actualReplyLength and actualReplicaSpecificInfoLength right after the call to .execute()

Reply:
Done.